### DISCUSSION TITLE
ASYNCIFY_ONLY doesn't decrease the link time process

### DISCUSSION THREAD

**Discussion by aharondavid** [emscripten-core/emscripten#18131] [Q&A]
Hi,
By using ACYNCIFY_ONLY=@file.txt that contain about 10 methods,  I accepted to decrease the link time process, but, unfortunately, it doesn't.
any one can explain?

**Comment by kripken** [comment]
Most of the work in Asyncify is just processing the wasm to a form that Asyncify can handle (flattening + optimizing). So instrumenting fewer functions will not make much difference there, I'm afraid.

The full solution is wasm stack switching, which will have normal/fast link times. It is [already testable](https://github.com/emscripten-core/emscripten/blob/e41f71699606d569970ad3501da8c53239468f3b/src/settings.js#L758-L765), but has not shipped yet in any browser.

edit: there is also an issue to only flatten what we need in Asyncify, https://github.com/WebAssembly/binaryen/issues/4822

**Comment by aharondavid** [comment]
Understand, thanks!.
